packaging_automation/prepare_release.py

```diff
@@ -0,0 +1,341 @@
+                                                                 config_py_path, configure_in_path,
+                                                                 devel_version, distributed_dir_path,
+                                                                 is_test, main_branch,
+                                                                 multi_extension_out_path,
+                                                                 multi_extension_sql_path,
+                                                                 project_name,
+                                                                 project_version, repository,
+                                                                 upcoming_minor_version,
+                                                                 upcoming_version,
+                                                                 upcoming_version_branch)
+        print(f"OK {project_version} Major release flow executed successfully")
+    else:
+        prepare_release_branch_for_patch_release(cherry_pick_enabled, configure_in_path, earliest_pr_date, is_test,
+                                                 main_branch, multi_extension_out_path, project_name, project_version,
+                                                 release_branch_name, repository)
+    return UpdateReleaseReturnValue(release_branch_name, upcoming_version_branch,
+                                    f"{DISTRIBUTED_DIR_PATH}/{newly_created_sql_file}")
+
+
+def prepare_release_branch_for_patch_release(cherry_pick_enabled, configure_in_path, earliest_pr_date, is_test,
+                                             main_branch, multi_extension_out_path, project_name, project_version,
+                                             release_branch_name, repository):
+    print(f"### {project_version} is a patch release. Executing Patch release flow ### ")
+    # checkout release branch (release-X.Y)
+    checkout_branch(release_branch_name, is_test)
+    # change version info in configure.in file
+    update_version_in_configure_in(configure_in_path, project_version)
+    # execute "auto-conf "
+    execute_autoconf_f()
+    # change version info in multi_extension.out
+    update_version_in_multi_extension_out(multi_extension_out_path, project_version)
+    if cherry_pick_enabled:
+        # cherry-pick the pr's with backport labels
+        cherrypick_prs_with_backport_labels(earliest_pr_date, main_branch, release_branch_name, repository)
+    # commit all changes
+    commit_changes_for_version_bump(project_name, project_version)
+    # create and push release-$minor_version-push-$curTime branch
+    release_pr_branch = f"{release_branch_name}_{uuid.uuid4()}"
+    create_and_checkout_branch(release_pr_branch)
+    if not is_test:
+        push_branch(release_pr_branch)
+
+
+def prepare_upcoming_version_branch(citus_control_file_path, config_py_path, configure_in_path, devel_version,
```

Comment:
So many parameters that are hard to keep track. Can we pass them in a param object?

Reply:
Fixed